app/models/search_facet.rb

```diff
@@ -309,80 +311,96 @@ def convert_time_between_units(base_value:, original_unit:, new_unit:)
       base_value
     else
       # first convert to seconds
-      value_in_seconds = self.calculate_time_in_seconds(base_value: base_value, unit_label: original_unit)
+      value_in_seconds = calculate_time_in_seconds(base_value: base_value, unit_label: original_unit)
       # now divide by multiplier to get value in new unit
       denominator = TIME_MULTIPLIERS[new_unit]
       value_in_seconds.to_f / denominator
     end
   end
 
-  # retrieve unique values from BigQuery and format an array of hashes with :name and :id values to populate :filters attribute
-  def get_unique_filter_values
-    Rails.logger.info "Updating filter values for SearchFacet: #{self.name} using id: #{self.big_query_id_column} and name: #{self.big_query_name_column}"
-    query_string = self.generate_bq_query_string
+  # retrieve unique values from BigQuery and format an array of hashes with :name and :id values to populate :filters
+  # can specify 'public only' to return filters for public studies
+  def get_unique_filter_values(public_only: false)
+    log_message = "Updating#{public_only ? ' public' : nil} filter values for SearchFacet: #{name} using id: " \
+                  "#{big_query_id_column} and name: #{big_query_name_column}"
+    Rails.logger.info log_message
+    if public_only
+      accessions = Study.where(public: true).pluck(:accession)
+      query_string = generate_bq_query_string(accessions: accessions)
+    else
+      query_string = generate_bq_query_string
+    end
     begin
       Rails.logger.info "Executing query: #{query_string}"
       results = SearchFacet.big_query_dataset.query(query_string)
-      self.is_numeric? ? results.first : results
+      is_numeric? ? results.first : results
     rescue => e
       Rails.logger.error "Error retrieving unique values for #{CellMetadatum::BIGQUERY_TABLE}: #{e.class.name}:#{e.message}"
-      ErrorTracker.report_exception(e, nil, { query_string: query_string})
+      ErrorTracker.report_exception(e, nil, { query_string: query_string, public_only: public_only })
       []
     end
   end
 
-  # update cached filters in place with new values
+  # update cached filters in place with new values, updating both public-only and regular list
+  # will update public-only values for non-numeric facets only since numeric facets have hard-coded ranges in UI
   def update_filter_values!
-    values = self.get_unique_filter_values
-    unless values.empty?
-      if self.is_numeric?
-        self.update(min: values[:MIN], max: values[:MAX])
-      else
-        self.update(filters: values.to_a)
-      end
+    if is_numeric?
+      values = get_unique_filter_values
+      return false if values.empty? # found no results, meaning an error occurred
```

Comment:
If an error occurred, should it be thrown and logged here?  I'd think so, but might be overlooking something.

Reply:
That is handled in `get_unique_filter_values`, which is where the actual call to BigQuery happens.  This is just the setter.